lib/models/NumericalDataSeries.js

```diff
@@ -3,19 +3,38 @@ import DataSeries from './DataSeries';
 class NumericalDataSeries extends DataSeries {
     constructor(data, options) {
         /*
-        *  Data is in the form of:
-        *     [{x_value: 1, y_value: 2}, {x_value: 3, y_value: 4}, etc]
-        */
+         *  Data is in the form of:
+         *     [{
+         *         x_value : 1,
+         *         x_min   : 0,
+         *         x_max   : 2,
+         *         y_value : 2,
+         *         y_min   : 1,
+         *         y_max   : 3
+         *     }, {
+         *         x_value : 3,
+         *         x_min   : 1,
+         *         x_max   : 4,
+         *         y_value : 4,
+         *         y_min   : 3,
+         *         y_max   : 5
+         *     }, etc]
+         *
+         *  Some common combination of fields are:
+         *      1. Range for X, Single value for Y - x_min, x_max, y_value (e.g., for a binned mean plot)
+         *      2. Single value for X, Range for Y - x_value, y_min, y_max (e.g., for an area plot)
+         *      3. Single value for X, Range + value for Y - x_value, y_min, y_max, y_value (e.g., for points with error bars)
+         */
```

Comment:
This is one of the main changes in this PR - I consolidated RangeDataSeries and NumericalDataSeries, since I didn't want an explosion of data series types depending on whether or not one of the axes supported a range. Otherwise we'd need XRangeYSingle, XSingleYRange, XRangeYRange etc. which would be nasty